### ISSUE TITLE
Authorization fails

### ISSUE THREAD

**Issue by ptwptw** [dreamfactorysoftware/javascript-sdk#2]
Trying to use the register/sign-in element of this demo, but get a "401. Invalid credentials supplied" error. I know the user details (username/password) are correct - I can see and set/check them in the Dreamfactory interface.

Looking at the ajax, it's doing a POST to:
https://<ACCOUNT>.enterprise.dreamfactory.com/api/v2/user/session

With params:
{"email":"myemail@not-a-real-domain.com","password":"mypassword"}

Perhaps there is extra code / extra config steps needed to get this working?

**Comment by todda48** [comment]
Make sure you are not trying to log in as an admin. It only allows login by users. README explains this and other setup info.

**Comment by ptwptw** [comment]
Okay - I've  checked further and potentially identified the issue ...

I was trying to login using a user account that existed in dreamfactory before the sample app was installed. (I installed the angular example, tested that, and then this one). I created a new role/permission for the new application. And I assigned that new role to the existing test user(s).

And I get "Invalid Credentials supplied". But I know they're correct.

Registering and logging in a new user works. 

The issue seems to be at the dreamfactory level:
When I edit a user, and from the role tab, select the role for this app, and then click "Update" - instead of updating, the role permission reverts to its original value immediately after the "Update" button is pressed. I don't think it's changing at all. Actually, there's seems to be some sort of ongoing bug here ... try changing user roles for an app and then clicking "update". Sometimes it works, sometimes it doesn't.
